### PR TITLE
Resolves Issue #103 noqa comments

### PR DESCRIPTION
**Pull request by tbrlpld**
Previously, `noqa` comments inside of a multi-line docstring where flagged as commented out code. When using `eradicate` directly on a file which includes such a `noqa` comment it does not show a violation. 

Example of such a multi-line docstring:
```python
def some_function():
    """
    Test for noqa comments in docstrings.

    This function has a multi-line doc string, but no return value is
    stipulated, while the function defines a return. This would raise DAR201
    flake8 violation. To suppress this raise violation the following noqa
    comment is defined in the docstring.

    # noqa: DAR201

    This noqa comment should not be detected as commented out code. `eradicate`
    itself does not raise this as a violation.

    """
    return "something"
```

This kind of `noqa` comments are needed to suppress violations raised by [`darglint`](https://github.com/terrencepreilly/darglint#ignoring-errors-in-a-docstring), e.g. violations associated with missing return statements in the docstring. 

This issue is caused by passing isolated physical lines of code to the `eradicate` function `filter_commented_out_code`. That function does expect to operate on a file's whole source code rather than on isolated lines of code. From the whole source code the context of each line can be determined. This allows differentiation between actual comment lines and lines that only look like comments. 

To resolve this issue, while still keeping the checker on one physical line at a time, a preliminary test is introduced that checks if the current physical line contains a comment according to the tokens of the physical line. These tokens are available as input to the plugin from `flake8`. 

Solves #103

### CODE REVIEW

File: flake8_eradicate.py [hunk 1]
```diff
@@ -73,12 +75,32 @@ def parse_options(cls, options) -> None:
...
-        return self._physical_line == filtered_source
+        Physical lines might appear like commented code although they are part
+        of a multi-line docstring (e.g. a `# noqa: DAR201` comment to suppress
+        flake8 warning about missing returns in the docstring).
+        To prevent this false-positive, the tokens of the physical line are
+        checked for a comment. The eradicate function is only invokes,
+        when the tokens indicate a comment in the physical line.
+
+        """
+        comment_in_line: bool = False
```
Comment (sobolevn) [inline flake8_eradicate.py]: you can remove the annotation here

Comment (tbrlpld) [reply flake8_eradicate.py]: Thank you for the feedback! 

I have removed the annotation in https://github.com/sobolevn/flake8-eradicate/pull/106/commits/8ee93a42ecfb065e0daa22d9a8695a5650807e3e

**Comment by sobolevn** [comment]
Thanks a lot, @tbrlpld 

P.S. Awesome commit messages! 👍

**Comment by tbrlpld** [comment]
@sobolevn Thank you 😄  

Always happy to help when I can.

**Comment by sobolevn** [comment]
I will release a new version tomorrow.

**Comment by tbrlpld** [comment]
Awesome! I am looking forward to using it 😄 👍

**Pull request merged**